pandas/tests/series/test_missing.py

```diff
@@ -1064,19 +1064,32 @@ def test_interp_limit(self):
         # GH 9217, make sure limit is an int and greater than 0
         methods = ['linear', 'time', 'index', 'values', 'nearest', 'zero',
                    'slinear', 'quadratic', 'cubic', 'barycentric', 'krogh',
-                   'polynomial', 'spline', 'piecewise_polynomial', None,
+                   'polynomial', 'spline', 'piecewise_polynomial',
                    'from_derivatives', 'pchip', 'akima']
         s = pd.Series([1, 2, np.nan, np.nan, 5])
         msg = (r"Limit must be greater than 0|"
                "time-weighted interpolation only works on Series or"
                r" DataFrames with a DatetimeIndex|"
-               r"invalid method '(polynomial|spline|None)' to interpolate|"
-               "Limit must be an integer")
+               r"Limit must be an integer|"
+               r"You must specify the order of the spline")
         for limit in [-1, 0, 1., 2.]:
             for method in methods:
                 with pytest.raises(ValueError, match=msg):
                     s.interpolate(limit=limit, method=method)
```

Comment:
Let's do same thing here: break up this test so that we can leverage `pytest` parameterization.

Reply:
I agree with you that this test ought to be broken up and simplified.

I will try to do that tonight, but if it proves difficult, I may propose doing it in a separate PR.

Reply:
Okay, ready for another look.